app/jobs/user_session_jobs/schedule_daily_messages_job.rb

```diff
@@ -0,0 +1,130 @@
+# frozen_string_literal: true
+
+class UserSessionJobs::ScheduleDailyMessagesJob < ApplicationJob
+  queue_as :question_sms
+
+  def perform(user_session_id)
+    user_session = UserSession.find(user_session_id)
+    user = user_session.user
+    session = user_session.session
+
+    # Find all accessible question groups
+    accessible_question_groups = calculate_accessible_question_groups_for_user(session.question_groups.order(:position), user_session)
+    today_scheduled_question_groups = select_questions_groups_scheduled_for_today(accessible_question_groups)
+
+    questions_to_be_send_today = []
+
+    # Find all questions scheduled for today in groups
+    today_scheduled_question_groups.each do |question_group|
+      last_answer = last_answer_in_question_group(question_group, user_session)
+      questions_per_day = question_group.sms_schedule['questions_per_day'] || 1
+      last_question_index = question_group.questions.order(position: :desc).first.position
+      first_question_index = question_group.questions.order(position: :asc).first.position
+
+      question_positions = if !last_answer || last_answer.question.position == last_question_index
+                             base_range = (first_question_index..last_question_index).to_a
+                             positions_range = base_range
+                             (questions_per_day / base_range.length.to_f).ceil.to_i.times do
+                               positions_range += base_range
+                             end
+                             positions_range
+                           elsif last_answer.question.position + questions_per_day > last_question_index
+                             base_range = (first_question_index..last_question_index).to_a
+                             from_current_question_range = ((last_answer.question.position + 1)..last_question_index).to_a
+                             ((questions_per_day / base_range.length.to_f).ceil - 1).to_i.times do
+                               from_current_question_range += base_range
+                             end
+                             from_current_question_range
+                           else
+                             ((last_answer.question.position + 1)..last_question_index).to_a
```

Comment:
Could you update the `Researcher: 6.d` requirement? Because this logic isn't defined in the task